src/mp4.cpp

```diff
@@ -38,6 +38,14 @@
 
 #include "src/impl.h"
 
+namespace mp4v2
+{
+   namespace impl
+   {
+      ShouldParseAtomCallback g_parseCallback = nullptr;
```

Comment:
Kind of wish this could be per instance of `MP4File`. But that is a pain.

I guess there's also thread safety to worry about, because some clients of mp4v2 will be running on different threads and presumably setting the `ShouldParseAtomCallback`.

Reply:
I don't think it's a pain, see my earlier comment

Reply:
Okay, I'll:
* fix the bug that Dave spotted right away
* make the callback per-instance
* remove zombie atoms

Another approach might be adding a setting to make parsing more lenient... log the error but don't throw an exception that fails `MP4Read()` entirely. That seems like a nicer fix but also bigger and more invasive.